source/simulator/helper_functions.cc

```diff
@@ -23,7 +23,16 @@
 #include <aspect/melt.h>
 #include <aspect/global.h>
 
+#include <aspect/geometry_model/interface.h>
+#include <aspect/heating_model/interface.h>
 #include <aspect/heating_model/adiabatic_heating.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/particle/generator/interface.h>
+#include <aspect/particle/integrator/interface.h>
+#include <aspect/particle/interpolator/interface.h>
+#include <aspect/particle/output/interface.h>
+#include <aspect/prescribed_stokes_solution/interface.h>
+#include <aspect/postprocess/visualization.h>
```

Comment:
That's not quite true. For example, all of the particle headers are not included in `simulator.h`, nor are the viz header file. You are correct about the prescribed_stokes_solution header, though, as well as possibly a few of the others.

In general, I think it's useful to really include in a file what is needed, rather than relying on some other header doing so transitively. What do you want me to do?